internal/store/store.go

```diff
@@ -202,12 +201,18 @@ func (s *LocalStore) Write(mdl v1.Image, tags []string, w io.Writer) error {
 			pr = progress.NewProgressReporter(w, progress.PullMsg, layer)
 			progressChan = pr.Updates()
 		}
-		if err := s.writeBlob(layer, progressChan); err != nil {
+
+		err := s.writeBlob(layer, progressChan)
+
+		if progressChan != nil {
 			close(progressChan)
-			return fmt.Errorf("writing blob: %w", err)
+			if err := pr.Wait(); err != nil {
+				fmt.Printf("waiting for reporter: %v\n", err)
```

Comment:
More context on this message might be helpful (indicating it's an error and not a blocking condition).

In general, it would be nice if `LocalStore` had a logger instead of printing directly to the console.

Reply:
Changed it a bit in https://github.com/docker/model-distribution/compare/b3f1b42895fa5c396268fe4069d47fc3cad4d2b6..1884ba6cf1129724f8f9fe56babed32581b40bb8. I don't have a lot of context. Maybe it's worth adding the layer ID, size as well?